src/ripple/protocol/IOUAmount.h

```diff
@@ -0,0 +1,129 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_PROTOCOL_IOUAMOUNT_H_INCLUDED
+#define RIPPLE_PROTOCOL_IOUAMOUNT_H_INCLUDED
+
+#include <beast/utility/noexcept.h>
+#include <beast/utility/Zero.h>
+#include <boost/operators.hpp>
+#include <cstdint>
+
+namespace ripple {
+
+class IOUAmount
+    : private boost::totally_ordered<IOUAmount>
+    , private boost::additive <IOUAmount>
+{
+private:
+    std::int64_t mantissa_;
+    int exponent_;
+
+    void
+    normalize ();
+
+public:
+    IOUAmount () = default;
+
+    IOUAmount (IOUAmount const& other) = default;
+
+    IOUAmount (beast::Zero zero)
+    {
+        *this = zero;
+    }
+
+    IOUAmount (std::int64_t mantissa, int exponent)
+        : mantissa_ (mantissa)
+        , exponent_ (exponent)
+    {
+        normalize ();
+    }
+
+    IOUAmount&
+    operator= (beast::Zero)
+    {
+        // The -100 is used to allow 0 to sort less than a small positive values
+        // which have a negative exponent.
+        mantissa_ = 0;
+        exponent_ = -100;
```

Comment:
If you are going to use the `-100` trick it needs to be a constant, because you'll need it elsewhere.